README.md

```diff
@@ -342,6 +343,36 @@ func SwapRESTConfig(rc *rest.Config) *reconcilers.SubReconciler {
 }
 ```
 
+#### WithFinalizer
+
+[`WithFinalizer`](https://pkg.go.dev/github.com/vmware-labs/reconciler-runtime/reconcilers#WithFinalizer) allows external state to be allocated and then cleaned up once the parent resource is deleted. When the parent resource is not terminating, the finalizer is set on the parent resource before the nested reconciler is called. When the parent resource is terminating, the finalizer is cleared only after the nested reconciler returns without an error.
```

Comment:
**[suggestion]:** Can you motivate usage of a `WithFinalizer` subreconciler over a `ChildReconciler/SyncRenciler` with `Finalize`?

Reply:
added a note under the finalizers section. We can continue to iterate on it